src/bessel.jl

```diff
@@ -7,37 +7,43 @@ struct AmosException <: Exception
 end
 
 ## Airy functions
-let
-    const ai::Array{Float64,1} = Array{Float64}(2)
-    const ae::Array{Int32,1} = Array{Int32}(2)
-    global _airy, _biry
-    function _airy(z::Complex128, id::Int32, kode::Int32)
-        ccall((:zairy_,openspecfun), Void,
-              (Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32},
-               Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32}),
-              &real(z), &imag(z),
-              &id, &kode,
-              pointer(ai,1), pointer(ai,2),
-              pointer(ae,1), pointer(ae,2))
-        if ae[2] == 0 || ae[2] == 3
-            return complex(ai[1],ai[2])
-        else
-            throw(AmosException(ae[2]))
-        end
+function _airy(z::Complex128, id::Int32, kode::Int32)
+    rz1, rz2 = Ref(real(z)), Ref(imag(z))
+    rid = Ref(id)
+    rkode = Ref(kode)
+    ai1, ai2 = Ref{Float64}(), Ref{Float64}()
+    ae1, ae2 = Ref{Int32}(), Ref{Int32}()
+
+    ccall((:zairy_,openspecfun), Void,
+          (Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32},
+           Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32}),
+           rz1, rz2, rid, rkode,
+           ai1, ai2, ae1, ae2)
+
+    if ae2[] == 0 || ae2[] == 3 # ignore underflow and less than half machine accuracy loss
+        return complex(ai1[], ai2[])
+    else
+        throw(AmosException(ae2[]))
     end
-    function _biry(z::Complex128, id::Int32, kode::Int32)
-        ccall((:zbiry_,openspecfun), Void,
-              (Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32},
-               Ptr{Float64}, Ptr{Float64}, Ptr{Int32}),
-              &real(z), &imag(z),
-              &id, &kode,
-              pointer(ai,1), pointer(ai,2),
-              pointer(ae,1))
-        if ae[1] == 0 || ae[1] == 3  # ignore underflow
-            return complex(ai[1],ai[2])
-        else
-            throw(AmosException(ae[2]))
-        end
+end
+
+function _biry(z::Complex128, id::Int32, kode::Int32)
+    rz1, rz2 = Ref(real(z)), Ref(imag(z))
+    rid = Ref(id)
+    rkode = Ref(kode)
+    ai1, ai2 = Ref{Float64}(), Ref{Float64}()
+    ae1 = Ref{Int32}()
+
+    ccall((:zbiry_,openspecfun), Void,
+          (Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32},
+           Ptr{Float64}, Ptr{Float64}, Ptr{Int32}),
```

Comment:
These should be `Ref`, not `Ptr`.  (Using `Ptr` here is probably a relic of the days before `Ref`.)

Once you change it to `Ref`, then you will no longer need to explicitly create `Ref` objects for parameters like `kode` that are purely inputs.  You can just pass `kode` directly, and `ccall` will create the `Ref` for you.